.teamcity/src/main/kotlin/projects/GradleBuildToolRootProject.kt

```diff
@@ -0,0 +1,32 @@
+package projects
+
+import common.Branch
+import jetbrains.buildServer.configs.kotlin.v2019_2.DslContext
+import jetbrains.buildServer.configs.kotlin.v2019_2.Project
+import model.CIBuildModel
+import model.JsonBasedGradleSubprojectProvider
+import model.StatisticBasedFunctionalTestBucketProvider
+import promotion.PromotionProject
+import util.UtilPerformanceProject
+import util.UtilProject
+import java.io.File
+
+class GradleBuildToolRootProject(branch: Branch) : Project({
+    id = DslContext.projectId
+    name = DslContext.projectName
+
+    val model = CIBuildModel(
+        projectId = "${DslContext.parentProjectId}_${branch.buildTypeId}",
+        branch = branch,
+        buildScanTags = listOf("Check"),
+        subprojects = JsonBasedGradleSubprojectProvider(File("./subprojects.json"))
+    )
+    val gradleBuildBucketProvider = StatisticBasedFunctionalTestBucketProvider(model, File("./test-class-data.json"))
+    subProject(CheckProject(model, gradleBuildBucketProvider))
+
+    subProject(PromotionProject(model.branch))
+    if (model.branch == Branch.Master) {
```

Comment:
I think we need the Util project on the other branches as well, given that they could contain a separate set of parameters.